controller/src/main/java/io/pravega/controller/server/rpc/grpc/GuardianAuthorizer.java

```diff
@@ -0,0 +1,48 @@
+package io.pravega.controller.server.rpc.grpc;
+
+import com.emc.nautilus.auth.client.GuardianClient;
+import com.emc.nautilus.auth.client.GuardianClientFactory;
+import com.emc.nautilus.auth.models.acls.TargetedACLs;
+import com.google.common.base.Strings;
+import io.grpc.Metadata;
+import io.grpc.ServerCall;
+import io.grpc.ServerCallHandler;
+import io.grpc.ServerInterceptor;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+public class GuardianAuthorizer implements ServerInterceptor {
+    private final GuardianClientFactory guardianClientFactory;
+    private final ConcurrentMap<String, GuardianClient> clientMap;
+
+    public GuardianAuthorizer(String guardianIp) {
+        this.guardianClientFactory = new GuardianClientFactory(guardianIp);
+        clientMap = new ConcurrentHashMap<>();
+    }
+
+    @Override
+    public <ReqT, RespT> ServerCall.Listener<ReqT> interceptCall(ServerCall<ReqT, RespT> call, Metadata headers, ServerCallHandler<ReqT, RespT> next) {
+        String method = headers.get(Metadata.Key.of("method", Metadata.ASCII_STRING_MARSHALLER));
+
+        if (!Strings.isNullOrEmpty(method) && method.equals("guardian")) {
+            String userName = headers.get(Metadata.Key.of("userName", Metadata.ASCII_STRING_MARSHALLER));
+            String token = headers.get(Metadata.Key.of("token", Metadata.ASCII_STRING_MARSHALLER));
+
+            GuardianClient client;
+            synchronized (this) {
+                if (!clientMap.containsKey(token)) {
```

Comment:
Looking ahead, for caching, Overseer uses this simple approach to cache a particular token's lookup results for 5 minutes. https://asdstash.isus.emc.com/projects/NAUT/repos/platform/browse/guardian-client/src/main/java/com/emc/nautilus/auth/acls/GuardianAccessControlHandler.java#8,24,30

Nautilus tokens can last 2 hours, but can be revoked during a logout.